scripts/linux_containers_on_windows/build_linux_kernel.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+set -xe -o pipefail
+
+source ~/.profile
+
+LINUX_BASE_DIR=""
+OPENGCS_BASE_DIR=""
+
+function get_job_number (){
+    #
+    # Multiply current number of threads with a number
+    # Usage:
+    #   ./build_artifacts.sh --thread_number x10
+    #
+    multi="${1#*x}"
+    cores="$(cat /proc/cpuinfo | grep -c processor)"
+    result="$(expr $cores*$multi | bc)"
+    echo ${result%.*}
+}
+
+function apply_patches() {
+    #
+    # Apply some patches before building
+    #
+    linux_base_dir="$1"; shift
+    linux_msft_artifacts_destination_path="$1"; shift
+    kernel_version="$1"; shift
+    checkout_commit="$1"; shift
+    cherry_pick_commit="$1"; shift
+    opengcs_base_dir="$1"; shift
+    commits_list="$1"
+
+
+    if [[ ! -d "${linux_base_dir}/msft_linux_kernel" ]]; then
+        echo "Could not find the Linux Kernel repo to apply patches"
+        exit 1
+    else
+        pushd "${linux_base_dir}/msft_linux_kernel"
+    fi
+
+    # this will be the destination directory of the kernel artifact
+    output_dir_name="${linux_msft_artifacts_destination_path}/$(date +%Y%m%d)_${BUILD_ID}__msft-kernel_${kernel_version}"
+    mkdir -p "${output_dir_name}"
+
+    # last commit before pulling
+    echo $(git log --pretty=format:'%h' -n 1) > "${output_dir_name}/latest_kernel_commit.log"
+    echo "Linux kernel built on commit:"
+    cat "${output_dir_name}/latest_kernel_commit.log"
+
+    # tag for specific repo
+    branch="$(git rev-parse --abbrev-ref HEAD)"
+    if [[ "${branch}" == "master" ]]; then
+        git checkout "${checkout_commit}"
+        git reset --hard HEAD~20 #Note (papagalu): if the env is not clean, we have to revert all cherry-picks
```

Comment:
why do you need the HEAD~20 in particular?

Reply:
because afterwards we precisely cherry-pick 20 commits